pkg/internal/tests/cross-tests/tfwrite.go

```diff
@@ -16,115 +16,75 @@
 package crosstests
 
 import (
-	"fmt"
-	"io"
-	"sort"
-
-	"github.com/hashicorp/hcl/v2/hclwrite"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
-	"github.com/zclconf/go-cty/cty"
+
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/internal/tests/cross-tests/impl/hclwrite"
 )
 
-// Provider writes a provider declaration to an HCL file.
-//
-// Note that unknowns are not yet supported in cty.Value, it will error out if found.
-func writeProvider(out io.Writer, sch map[string]*schema.Schema, typ string, config cty.Value) error {
-	if !config.IsWhollyKnown() {
-		return fmt.Errorf("WriteHCL cannot yet write unknowns")
-	}
-	f := hclwrite.NewEmptyFile()
-	block := f.Body().AppendNewBlock("provider", []string{typ})
-	writeBlock(block.Body(), sch, config.AsValueMap())
-	_, err := f.WriteTo(out)
-	return err
-}
+// This is a copy of the NestingMode enum in the Terraform Plugin SDK.
+// It is duplicated here because the type is not exported.
+type sdkV2NestingMode int
 
-// Resource writes a resource declaration to an HCL file.
-//
-// Note that unknowns are not yet supported in cty.Value, it will error out if found.
-func writeResource(
-	out io.Writer, sch map[string]*schema.Schema, resourceType, resourceName string, config cty.Value,
-) error {
-	if !config.IsWhollyKnown() {
-		return fmt.Errorf("WriteHCL cannot yet write unknowns")
+const (
+	sdkV2NestingModeInvalid sdkV2NestingMode = iota
+	sdkV2NestingModeSingle
+	sdkV2NestingModeGroup
+	sdkV2NestingModeList
+	sdkV2NestingModeSet
+	sdkV2NestingModeMap
+)
+
+func sdkV2NestingToShim(nesting sdkV2NestingMode) hclwrite.Nesting {
+	switch nesting {
+	case sdkV2NestingModeSingle:
+		return hclwrite.NestingSingle
+	case sdkV2NestingModeList:
+		return hclwrite.NestingList
+	case sdkV2NestingModeSet:
```

Comment:
You're not handling sdkV2NestingModeMap with any warnings or such, worth a comment?

Reply:
I believe the other nesting modes are not used. I have added an assert there with a message.